compiler/rustc_query_impl/src/plumbing.rs

```diff
@@ -370,6 +381,34 @@ pub(crate) fn encode_query_results<'a, 'tcx, Q>(
     });
 }
 
+pub(crate) fn query_key_hash_verify<'tcx>(
+    query: impl QueryConfig<QueryCtxt<'tcx>>,
+    qcx: QueryCtxt<'tcx>,
+) {
+    let _timer =
+        qcx.profiler().generic_activity_with_arg("query_key_hash_verify_for", query.name());
+
+    let mut map = FxHashMap::default();
```

Comment:
Would you mind using an [UnordMap](https://doc.rust-lang.org/stable/nightly-rustc/rustc_data_structures/unord/struct.UnordMap.html) here? I think we should start phasing out the usage of raw FxHashMap in favor of FxIndexMap and UnordMap.